# Problem (a) Solution: One-Sample t-test

### 1. State the Hypotheses
- Null hypothesis ($H_0$): The mean part length equals the standard value.
  $H_0: \mu = 10.0 \, \text{cm}$
- Alternative hypothesis ($H_1$): The mean part length is significantly different from the standard value.
  $H_1: \mu \neq 10.0 \, \text{cm}$ (Two-tailed test)

In [1]:
# Given data
sample_mean = 10.06  # Sample mean (cm)
pop_mean = 10.0      # Hypothesized population mean (cm)
sample_std = 0.19    # Sample standard deviation (cm)
n = 25               # Sample size
alpha = 0.05         # Significance level

# Calculate test statistic (t-value)
import numpy as np
t_stat = (sample_mean - pop_mean) / (sample_std / np.sqrt(n))

# Calculate critical t-value
from scipy import stats
df = n - 1  # Degrees of freedom
t_critical = stats.t.ppf(1 - alpha/2, df)  # Two-tailed critical value

print(f'Test statistic (t): {t_stat:.4f}')
print(f'Degrees of freedom: {df}')
print(f'Critical t-values: ±{t_critical:.4f}')

Test statistic (t): 1.5789
Degrees of freedom: 24
Critical t-values: ±2.0639


### 2. Rejection Region
The rejection region for a two-tailed test with $\alpha = 0.05$ and $df = 24$ is:
$t < -2.0639$ or $t > 2.0639$

### 3. Conclusion
The calculated test statistic ($t = 1.5789$) falls within the non-rejection region ($-2.0639 < 1.5789 < 2.0639$).

We fail to reject the null hypothesis $H_0$. At the $\alpha = 0.05$ significance level, there is insufficient evidence to conclude that the mean part length differs from the standard value of 10.0 cm.